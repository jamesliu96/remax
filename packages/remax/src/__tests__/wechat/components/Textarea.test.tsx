import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Textarea from '../../../adapters/wechat/components/Textarea';

describe('TextArea', () => {
  it('render correctly', () => {
    const testRenderer = TestRenderer.create(
      <Textarea className="class" value="" />
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('focus correctly', () => {
    const component: TestRenderer.ReactTestRenderer = TestRenderer.create(
      <Textarea value="" />
    );

    const instance = component.root.findByType('textarea');

    expect(instance.props.focus).not.toBeTruthy();

    TestRenderer.act(() => {
      instance.props.onClick();
    });

    expect(instance.props.focus).toBeTruthy();

    TestRenderer.act(() => {
      instance.props.onBlur();
    });

    expect(instance.props.focus).not.toBeTruthy();
  });
});
