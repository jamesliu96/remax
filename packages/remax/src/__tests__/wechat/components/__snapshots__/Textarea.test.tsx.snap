// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea render correctly 1`] = `
<textarea
  className="class"
  fixed={false}
  focus={false}
  maxlength={-1}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onInput={[Function]}
  selectionEnd={999}
  selectionStart={999}
  value=""
/>
`;
